speaker/bgp_controller.go

```diff
@@ -228,6 +228,17 @@ func (c *bgpController) SetBalancer(l log.Logger, name string, lbIP net.IP, pool
 	c.svcAds[name] = nil
 	for _, adCfg := range pool.BGPAdvertisements {
 		m := net.CIDRMask(adCfg.AggregationLength, 32)
+		if lbIP.To4() == nil {
+			if c.bgpType == bgpNative {
```

Comment:
Here it might be too late. I'd do this check in https://github.com/metallb/metallb/blob/9cb1653a87673ec91971a60c873204e0752ff97b/speaker/main.go#L350 
where if we return an error the config gets staled and an alert is thrown.